app/pages/patient/patientinfo.js

```diff
@@ -374,12 +391,12 @@ var PatientInfo = React.createClass({
     }
 
     var birthday = formValues.birthday;
-    if (!(birthday && datetimeUtils.isValidDate(birthday))) {
+    if (!(birthday && sundial.isValidDate(birthday))) {
```

Comment:
so even according to the moment docs there parsing and subsequent date validation is very loose and hence our use of the isValid can yield a result that isn't strictly valid. 

W can also use a strict parsing which requires that the format and input match exactly and as in this case we have both we can use that. Details http://momentjs.com/docs/#/parsing/string-format/